includes/wikia/parser/templatetypes/TemplateTypesParser.class.php

```diff
@@ -35,6 +36,71 @@ public static function onFetchTemplateAndTitle( &$text, &$finalTitle ) {
 	}
 
 	/**
+	 * @desc change template wikitext according to template type
+	 *
+	 * @param $templateTitle
+	 * @param $templateWikitext
+	 * @return bool
+	 */
+	public static function onBraceSubstitution( $templateTitle, &$templateWikitext ) {
+		global $wgEnableTemplateTypesParsing, $wgArticleAsJson, $wgCityId;
+
+		wfProfileIn( __METHOD__ );
+
+		if ( $wgEnableTemplateTypesParsing
+			&& $wgArticleAsJson
+			&& !empty( $templateWikitext )
+			&& $title = self::getValidTitle( $templateTitle ) ) {
+			$type = ( new ExternalTemplateTypesProvider( new \TemplateClassificationService ) )
+				->getTemplateTypeFromTitle( $wgCityId, $title );
+
+			if ( $type == AutomaticTemplateTypes::TEMPLATE_CONTEXT_LINK ) {
+				$templateWikitext = self::handleContextLinksTemplate( $templateWikitext );
+			}
+		}
+
+		wfProfileOut( __METHOD__ );
+
+		return true;
+	}
+
+	/**
+	 * @desc sanitize context-link template content - remove all non-link and non-text
+	 * elements from context-link template output and wrap it in div with special class
+	 *
+	 * @param $templateWikitext
+	 * @return string
+	 */
+	public static function handleContextLinksTemplate( $templateWikitext ) {
+		//remove any custom HTML tags
+		$templateWikitext = strip_tags( $templateWikitext );
+		//remove list and indent elements from the beginning of line
+		$templateWikitext = preg_replace( '/^[:#* \n]+/', '', $templateWikitext );
+		//remove all bold and italics from all of template content
+		$templateWikitext = preg_replace( '/\'{2,}/', '', $templateWikitext );
+		//remove all newlines from the middle of the template text.
+		$templateWikitext = preg_replace( '/\n/', ' ', $templateWikitext );
+		//wrap text of context-link in specified class
+		$templateWikitext = '<div class="' . self::CLASS_CONTEXT_LINK . '">' . $templateWikitext . '</div>';
+
+		return $templateWikitext;
+	}
+
+	/**
+	 * @desc return a valid title from template title got from Parser
+	 * or false if invalid
+	 *
+	 * @param $title
+	 * @return bool|\Title
+	 * @throws \MWException
+	 */
+	private static function getValidTitle( $title ) {
```

Comment:
This method name and params are confusing. I think that $titleText would be more explicit, and at the same time I'd stick to single return type ( isValidTitle )